Refactor RemoteAPI users to share more code [AP-2069]

chouquette opened the pull request:
<!--
* New contributors are highly encouraged to read our
  [CONTRIBUTING](/CONTRIBUTING.md) documentation.
* The pull request:
  * Should only fix one issue or add one feature at a time.
  * Must update the test suite for the relevant functionality.
  * Should pass all status checks before being reviewed or merged.
* Commit titles should be prefixed with general area of pull request's change.
* Draft PRs should be prefixed with `[WIP]` in their title.

-->
### What does this PR do?

This PR refactors the usage of RemoteAPI to use a single common implementation of `make_request` instead of having a duplicated version with minor differences in all subclasses.

<!--
* A brief description of the change being made with this pull request.
* If the description here cannot be expressed in a succinct form, consider
  opening multiple pull requests instead of a single one.
-->

### Motivation

AP-2069

<!--
* What inspired you to submit this pull request?
* Link any related GitHub issues or PRs here.
-->

### Additional Notes

<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

### Possible Drawbacks / Trade-offs

<!--
* What are the possible side-effects or negative impacts of the code change?
-->

### Describe how to test/QA your changes

<!--
* Write here in detail or link to detailed instructions on how this change can
  be tested/QAd/validated, including any environment setup.
-->

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

chouquette committed:
tasks: githubapp: use common RemoteAPI code to issue requests [AP-2069]

chouquette committed:
tasks: refactor RemoteAPI users to share more code [AP-2069]

All the RemoteAPI child class now use make_request the same way, only to
insert their authentication header, so we can simply craft the header in
the per-platform code and use the same make_request helper in common
code

chouquette committed:
tasks: githubapp: inherit from RemoteAPI

chouquette commented on tasks/libs/common/github_api.py:
```diff
@@ -76,27 +76,11 @@ def get_milestone_by_name(self, milestone_name):
-            stream_output=False,
-            method=method,
-        )
+    def auth_headers(self):
```
I dislike the idea of using API specific headers as the default, but we could reach a similar result by having GithubWorkflow inheriting from GithubAPI (I'm not sure about the GithubApp class yet)

chouetz replied:
ok then no need to do more, on my side I'm not fan of spending time on our own implementation of such api

chouquette replied:
Agreed, I just saw your ticket to switch to an official implementation, I might take a look at it :)

chouquette committed:
tasks: GithubWorkflow: inherit from github_api to share more code

chouquette committed:
githubapp: fix BASE_URL assignment

RemoteAPI uses BASE_URL, not base_url. Initialize the class variable as
it is done for other RemoteAPI implementations

chouquette committed:
githubapp: fix make_request calls

chouquette committed:
gitlab: apply black

chouquette committed:
githubapp: apply black

chouquette committed:
apply isort

chouquette commented:
Closing in favor of https://github.com/DataDog/datadog-agent/pull/18592
